doc/specification/language-proposal.md

```diff
@@ -0,0 +1,1518 @@
+3. Left hand side of `effect`, corresponding to contributions of the interface to the simulation state.
+4. Left hand side of `initial`, where the optional `... from` clause modifies the initial state, and also where `state` forms the left hand side of the final binding.
+5. Left hand side of `evolve`, where the keyword `explicit` might be replaced with other possible descriptions, such as `implicit` for implicit ODE or DAE systems, and also where `state'` forms left hand side of the final binding.
+6. Left hand side of `export`, where we have `density parameter` or `parameter`, but which could be extended to support for example the exposure of derived values to probe requests or similar.
+
+In addition, the set of possible interface classes can be extended. An example would be to add stateful gap junctions.
+
+## Default modules
+
+Rather than having functions such as `exp` and constant such as `π` bound by default in module scopes, built-in functions and constants could be moved to a module that is always provided by the environment. This module name can be freely chosen, but suggestions have included: `common`, `std`, `prelude`. `arblang` is another possibility.
+
+Importing many constants or functions from a module may be tedious: either the qualified name is used, or each must be introduced via a `def` or `let` binding. A possible extension to Arblang would be to permit `with M;` to introduce local bindings for all constants and functions defined in a module imported as _M_, in an expression scope.
+
+Possible example:
+```
+module foo {
+    import common
+
+    # Using a qualified name:
+    def a = common.cos(common.π/5);
+
+    # Explicit bindiings:
+    def cos = common.cos;
+    def π = common.π;
+    def b = cos(π/5);
+
+    # Using `with`:
+    def c = with common; cos(π/5);
+}
+```
+
+## Alternative concentration model
+
+The concentration models described in [Mechanism semantics](#mechanism-semantics) are defined in terms of flow contributions, rather than in absolute concentrations, which is a departure from the NEURON NMODL approach.
+
+The NEURON approach has some limitations: initial concentration must be supplied by the mechanism model, not the cell description; and all contributions to concentration evolution must be combined within the one mechanism, because different concentration writing mechanisms cannot overlap. Nonetheless, the flow based description
+
+```
+interface concentration "CaDynamics" {
+    export parameter gamma: real = 0.5;
+    export parameter decay: time = 80 ms;
+    export parameter detph: length = 0.1 µm;
+    export parameter steady_conc: concentration = 1.0e-4 mmol/L;
+
+    bind ca_flux = molar flux "ca";
+    bind ca_conc = internal concentration "ca";
+    effect internal concentration rate "ca" = -ca_flux*gamma/depth - (ca_conc-steady_conc)/decay;
+}
+```
+
+might be represented in a direct concentration model as
+
+```
```

Comment:
Didn't we agree that this shouldn't be allowed? Why would we give it as an option?

Reply:
I wasn't sure we were all sure on that actually. I'll happily remove it.